tools/nightly-e2e-tests/monitors-gen/templates/monitor-apm.tf.src

```diff
@@ -0,0 +1,25 @@
+resource "datadog_monitor" ${{monitor_id}} {
+  name               = ${{monitor_name}}
+  type               = "query alert"
+  tags               = ["service:com.datadog.ios.nightly", "env:instrumentation", "team:rumm", "source:ios", ]
+  message            = <<EOT
+@maciek.grzybowski@datadoghq.com
+@mert.buran@datadoghq.com
+EOT
+  escalation_message = <<EOT
+<nil>
+EOT
+
+  query               = ${{monitor_query}}
+  notify_no_data      = ${{notify_no_data:-false}}
```

Comment:
shouldn't we make it `true` by default? because majority of the tests expect data

Reply:
TBH I didn't put too much attention to this value initially. Among all 210 Android monitors that I initially exported none was defining `notify_no_data = true` 🤔, so I didn't customize this value for any iOS monitor definition.

The [doc says](https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/monitor#optional):
> * `notify_no_data` (Boolean) A boolean indicating whether this monitor will notify when data stops reporting. Defaults to `false`.

so indeed, sounds like this will raise an alert if monitor has no data. On the other side, if there is no data the monitor will trigger anyway - as its assertion won't be met. That said, I think that this value is redundant to what we already observe, but I will try with `true` for the sake of correctness 👌.

Reply:
Done in https://github.com/DataDog/dd-sdk-ios/pull/521/commits/2c571d18f7ac7f91c0db91ecbf1a54f881120952 👍

Reply:
I didn't check it in the export, but majority of the monitors I've created have this:

![image](https://user-images.githubusercontent.com/4046447/122392544-407caa80-cf74-11eb-976c-e196ec6923c5.png)

Maybe `dd2tf` is missing something in the export template? I'm not sure if monitor will be triggered if there is no data, but I didn't check if it happens or not tbh.

Reply:
Which made me realise that we also have to use `no_data_timerframe` set to `24h` - done in https://github.com/DataDog/dd-sdk-ios/pull/521/commits/b217d6ea50e78202d5904d84d91f7b8cdd891f00 thanks 🙂.

About `dd2tf` - it is not very up-to-date and, so maybe that's why (I can't find `no_data_timerframe` there neither).